File: Hashmap/LC128_longest_consecutive_sequence.java
class Solution {
    public int longestConsecutive(int[] nums) {
        HashSet<Integer> set = new HashSet<Integer>();
        for (int num: nums) set.add(num);
        int longest = 0;
        
        for (int num: set) {
            if (!set.contains(num - 1)) {
                int cur = num;
                int count = 1;
                
                while (set.contains(++cur)) {
                    count ++;
                }
                
                longest = Math.max(longest, count);
            }
        }
        
        return longest;
    }
}

File: Union Find/LC128-longest-consecutive-sequence.py
class Solution:
    ans = 1
    def longestConsecutive(self, nums: List[int]) -> int:
        parent = {}
        size = {}
        
        def find(x):
            p = parent.get(x, x)
            if p != x:
                parent[x] = find(p)
            return parent.get(x, x)
        
        def union(x, y):
            px, py = find(x), find(y)
            if px != py:
                sx, sy = size.get(px, 1), size.get(py, 1)
                if sx < sy:
                    px, py = py, px
                parent[py] = px
                size[px] = sx + sy
            self.ans = max(size.get(px, 1), self.ans)
        if len(nums) == 0:
            return 0
        s = set(nums)
        for i in range(len(nums)):
            cur = nums[i]
            if cur - 1 in s:
                union(cur, cur-1)
            if cur + 1 in s:
                union(cur, cur+1)
        
        return self.ans